source/common/buffer/zero_copy_input_stream_impl.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "envoy/buffer/buffer.h"
+
+#include "google/protobuf/io/zero_copy_stream.h"
+
+namespace Envoy {
+
+namespace Buffer {
+
+class ZeroCopyInputStreamImpl : public virtual google::protobuf::io::ZeroCopyInputStream {
```

Comment:
Is `virtual` needed here? Is this because `google::protobuf::io::ZeroCopyInputStream` isn't pure? Do have potential diamond inheritance with this?